### ISSUE TITLE
salt-call does not respect/honor CLI `test=False` while minion uses config `test: True`

### ISSUE THREAD

**Issue by uvsmtid** [saltstack/salt#10877]
When test mode is set by default (in minion config `test: True`), at least `salt-call --local` command does not respect override given in command line.

### Actual Results

```
> grep 'test:' /etc/salt/minion
test: True

> salt-call --local state.highstate test=False
...
Summary
-------------
Succeeded: 31
Failed:     0
Not Run:    3
-------------
Total:     34
```
Some states were not run.

### Expected Result

The state should be executed as if config file doesn't have `test: True` setting:
```
> grep 'test:' /etc/salt/minion
# no output

> salt-call --local state.highstate test=False
....
Summary
-------------
Succeeded: 34
Failed:     0
-------------
Total:     34
```
All states were run.

### Note

This issue correlates with: #7788, #5411, #3363.

The combination where config `test: False` and CLI `test=True` is fine: CLI overrides config and runs all states in test mode as expected.

```
> grep 'test:' /etc/salt/minion
test: False

> salt-call --local state.highstate test=True
...
Summary
-------------
Succeeded: 31
Failed:     0
Not Run:    3
-------------
Total:     34
```
### Version
```
> salt --versions-report
           Salt: 2014.1.0
         Python: 2.7.5 (default, Feb 19 2014, 13:47:28)
         Jinja2: 2.7.1
       M2Crypto: 0.21.1
 msgpack-python: 0.1.13
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
         PyYAML: 3.10
          PyZMQ: 13.0.2
            ZMQ: 3.2.4

> rpm -qf $(which salt-call)
salt-minion-2014.1.0-1.fc20.noarch
```

**Comment by cachedout** [comment]
This is broken on 2015.5 as well.

**Comment by cachedout** [comment]
I have tested this against 2014.7 and 2015.5. In both cases, the behaviour for salt-call is correct. Since we aren't pushing fixes into 2014.1 any longer, please upgrade to 2014.7 or higher and see if this problem is resolved for you. Thanks.
